### PR TITLE
Attempt to add a way to introspect the names available for fetch

### PR DESCRIPTION
**Pull request by Cadair**
This also fixes a spelling mistake of `phyobs` to `physobs` and adds a section on the `path` argument to fetch.

### CODE REVIEW

File: docs/guide/acquiring_data/fido.rst [hunk 1]
```diff
@@ -107,8 +107,24 @@ Downloading data
 Once you have located your files via a ```Fido.search```, you can download
 them via ```Fido.fetch```::
 
-    >>> downloaded_files = Fido.fetch(result)
+    >>> downloaded_files = Fido.fetch(results)
```
Comment (SolarDrew) [inline docs/guide/acquiring_data/fido.rst]: Was it not decided 'fetch'es are changing to `get`s? I may have misremembered this.

Comment (Cadair) [inline docs/guide/acquiring_data/fido.rst]: other way around...

Comment (nabobalis) [inline docs/guide/acquiring_data/fido.rst]: It best be.

File: sunpy/net/dataretriever/client.py [hunk 2]
```diff
@@ -187,7 +198,12 @@ def _makeargs(self, *args, **kwargs):
                 if a_min == a_max:
                     self.map_[elem.__class__.__name__.lower()] = a_min
                 else:
-                    self.map_[elem.__class__.__name__.lower()] = (a_min, a_max)
+                    if elem.__class__ is Wavelength:
```
Comment (SolarDrew) [inline sunpy/net/dataretriever/client.py]: I'm sure it doesn't matter, but why not `isinstance` here?

**Comment by SolarDrew** [comment]
👍

File: docs/guide/acquiring_data/fido.rst (hunk 1 above)
Comment (SolarDrew) [inline docs/guide/acquiring_data/fido.rst]: Fair enough, my bad.

Comment (Punyaslok) [inline docs/guide/acquiring_data/fido.rst]: iirc it was changed to `download`

File: sunpy/net/dataretriever/client.py [hunk 3]
```diff
@@ -44,15 +44,15 @@ def __init__(self, map0, url, time=None):
...
         self.source = map0.get('source', "Data not Available")
         self.provider = map0.get('provider', "Data not Available")
-        self.phyobs = map0.get('phyobs', "Data not Available")
+        self.physobs = map0.get('physobs', "Data not Available")
         self.instrument = map0.get('instrument', "Data not Available")
         self.url = url
         self.time = TimeRange(map0.get('Time_start'),
                               map0.get('Time_end')) if time is None else time
-        self.wavelength = map0.get('wavelength', np.NaN)
+        self.wave = map0.get('wavelength', np.NaN)
```
Comment (Cadair) [inline sunpy/net/dataretriever/client.py]: @Punyaslok this will mess with #1830

File: sunpy/net/dataretriever/client.py (hunk 2 above)
Comment (Cadair) [inline sunpy/net/dataretriever/client.py]: fixed.

File: sunpy/net/dataretriever/client.py (hunk 3 above)
Comment (Punyaslok) [inline sunpy/net/dataretriever/client.py]: Merge this first, then whatever fails will be fixed

**Pull request merged**